tests/functional/tests/security/test_metadata_corruption.py

```diff
@@ -0,0 +1,267 @@
+
+    primary_cache_volume = RamVolume(Size.from_MiB(60))
+    secondary_cache_volume = RamVolume(Size.from_MiB(60))
+
+    core_volume = RamVolume(Size.from_MiB(1))
+
+    secondary_cache = Cache(
+        owner=secondary_ctx,
+        cache_mode=cache_mode,
+        cache_line_size=cache_line_size,
+    )
+    secondary_cache.start_cache()
+    secondary_cache.standby_attach(secondary_cache_volume)
+
+    corrupted_bytes = get_random_target_in_segment(secondary_cache, target_segment)
+
+    secondary_cache_exp_obj = CacheVolume(secondary_cache)
+    primary_cache_replicated_volume = ReplicatedVolume(
+        primary_cache_volume, secondary_cache_exp_obj
+    )
+
+    primary_cache = Cache.start_on_device(
+        primary_cache_replicated_volume,
+        owner=primary_ctx,
+        cache_mode=cache_mode,
+        cache_line_size=cache_line_size,
+    )
+    core = Core(core_volume)
+    primary_cache.add_core(core)
+
+    core_exp_volume = CoreVolume(core)
+    queue = primary_cache.get_default_queue()
+
+    r = (
+        Rio()
+        .target(core_exp_volume)
+        .njobs(1)
+        .readwrite(ReadWrite.WRITE)
+        .size(Size.from_MiB(1))
+        .qd(1)
+        .run([queue])
+    )
+
+    if recovery:
+        primary_cache.save()
+        primary_cache.device.offline()
+
+    exc = False
+    try:
+        primary_cache.stop()
+    except OcfError:
+        exc = True
+
+    primary_cache_replicated_volume.online()
+
+    secondary_cache.stop()
+
+    assert recovery == exc, "Stopping with device offlined should raise an exception"
+
+    for byte in corrupted_bytes:
+        corrupt_byte(secondary_cache_volume.data, corrupted_bytes)
+
+    loaded = False
+    with expectation:
+        secondary_cache = Cache.load_standby_from_device(
+            secondary_cache_volume, secondary_ctx, cache_line_size=cache_line_size
+        )
+        loaded = True
+
+    if loaded:
+        secondary_cache.standby_detach()
+
+        secondary_cache.standby_activate(secondary_cache_volume, open_cores=False)
+
+
+
+def corrupt_byte(buffer, offset):
+    logger.info(f"Corrupting byte {offset}")
+    byte_val = int.from_bytes(buffer[offset], "big")
+    target_val = byte_val ^ 0xAA
+    buffer[offset] = (target_val).to_bytes(1, "big")
+    logger.debug(f"0x{byte_val:02X} -> 0x{target_val:02X}")
+
+
+def get_random_target_in_segment(cache: Cache, target_segment: CacheMetadataSegment):
+    offset = Size.from_page(get_metadata_segment_page_location(cache, target_segment))
+    page_count = get_metadata_segment_size(cache, target_segment)
+    if get_metadata_segment_is_flapped(cache, target_segment):
+        page_count = page_count // 2
+    elem_size = get_metadata_segment_elem_size(cache, target_segment)
+    elems_per_page = get_metadata_segment_elems_per_page(cache, target_segment)
+    elems_count = get_metadata_segment_elems_count(cache, target_segment)
+
+    target_page = randrange(0, page_count) if page_count > 1 else 0
```

Comment:
Is the `if` statement really necessary? Min value of `page_count` is 1 and `randrange(0,1)` always return 0 which is correct.